### PR TITLE
Verify Changelog Like ForRelease

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- If the version being verified has a date as the release status verify the changelog like it is being released. 
- Removed the strict date verification that check to make sure the date is within the current month. Now it just checks to make sure the date for the verified version is the latest in the CHANGELOG file.
-This should finally resolve https://github.com/Azure/azure-sdk/issues/3690

### CODE REVIEW

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (weshaggard) [inline eng/common/scripts/ChangeLog-Operations.ps1:326]:
```suggestion
function Confirm-ChangeLogForRelease {
```

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (weshaggard) [inline eng/common/scripts/ChangeLog-Operations.ps1:349]: We should probably use a common sort helper method so we don't have differences in how we sort here vs in the update.

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (weshaggard) [inline eng/common/scripts/ChangeLog-Operations.ps1:390]: This message doesn't seem correct.

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (chidozieononiwu) [reply eng/common/scripts/ChangeLog-Operations.ps1:349]: This one is sorting only the dates though. If we want to have the same sorting, then we can just use the order of the changelog entries which is almost always already sorted by prior processes.

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (chidozieononiwu) [reply eng/common/scripts/ChangeLog-Operations.ps1:390]: This is because I am reporting all the various errors that occur before exiting the validation process. Previously we just exited immediately an error occurs

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (weshaggard) [reply eng/common/scripts/ChangeLog-Operations.ps1:349]: I think we should use the same sorting as the change log entries to just be sure they are consistent. Currently we sort the entries right before we save them so I just want to be sure that we are consistent.

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (weshaggard) [inline eng/common/scripts/ChangeLog-Operations.ps1:361]: Do we need to trim everything here? Can't we just do something like `$entries.ReleaseStatus[0] -ne $changeLogEntry.ReleaseSTatus`?

File: eng/common/scripts/ChangeLog-Operations.ps1
Comment (weshaggard) [reply eng/common/scripts/ChangeLog-Operations.ps1:402]: Shouldn't this message be removed?

**Review by weshaggard — APPROVED** [review]
A couple clean-up comments but otherwise looks good. Lets be sure to run this on all the changelogs in our primary repos before we merge this change and we should hold off until after release week next week as well.

**Comment by weshaggard** [comment]
@chidozieononiwu did you complete the validation by running it on all the changelogs in our primary repos?

**Comment by chidozieononiwu** [comment]
> @chidozieononiwu did you complete the validation by running it on all the changelogs in our primary repos?

I did. There were minor issues which I have fixed on other PRs. All of them were due to errors in the version/changelog

**Pull request merged**
